tamr_toolbox/utils/custom_button.py

```diff
@@ -0,0 +1,490 @@
+
+    # Page name validation
+    invalid_pages = [p for p in page_names if not _check_valid_page_name(pagename=p)]
+    if len(invalid_pages) > 0:
+        value_error_message = (
+            f"Invalid pagename(s): {invalid_pages}. See docs for allowed Tamr button page names"
+        )
+        LOGGER.error(value_error_message)
+        raise ValueError(value_error_message)
+
+    button_dict = {
+        "extensionName": extension_name,
+        "buttons": [
+            {
+                "buttonType": "redirectButton",
+                "buttonId": button_id,
+                "buttonText": button_text,
+                "pageNames": page_names,
+                "redirectUrl": redirect_url,
+                "openInNewTab": open_in_new_tab,
+            }
+        ],
+    }
+
+    file = f"{button_name}.yaml"
+    filepath = os.path.join(output_dir, file)
+
+    LOGGER.info(f"Saving {file} to {output_dir}")
+    with open(f"{filepath}", "w") as yaml_file:
+        yaml.dump(button_dict, yaml_file, sort_keys=False)
+
+    return filepath
+
+
+def create_post_button(
+    *,
+    extension_name: str,
+    button_id: str,
+    button_text: str,
+    page_names: List[str],
+    post_url: str,
+    post_body_keys: List[str],
+    success_message: str,
+    fail_message: str,
+    display_response: bool,
+    output_dir: str,
+    button_name: str,
+) -> str:
+    """
+    Create yaml file with all required attributes for
+    a 'POST' UI button. Yaml file is saved locally.
+
+    Button features are only available to versions 2022.008.0 and later.
+
+    Args:
+        extension_name: Name of button extension
+        button_id: A short identifier for the button to use in the,
+                   body of a POST call or a redirect URL path substitution.
+        button_text: The button label to display in the UI.
+        page_names: The pages of the UI on which to display the button.
+        post_url: The target URL for a POST API call
+        post_body_keys: Specifies the keys to request in the body of the POST call
+        success_message: The message that displays to the user when the POST call succeeds.
+        fail_message: The message that displays to the user when the POST call fails.
+        display_response: Whether the contents of the API response body should display to the user.
+        output_dir: Directory to save yaml file (absolute path)
+        button_name: Name of yaml file
+
+    Returns:
+        Path to yaml file created
+    """
+    # Path validation
+    _check_valid_abs_path(output_dir)
+
+    # Minor url validation
+    if not post_url.startswith(("http://", "https://")):
+        value_error_message = f"Invalid url. Must begin with http:// or https://"
+        LOGGER.error(value_error_message)
+        raise ValueError(value_error_message)
+
+    # Page name validation
+    invalid_pages = [p for p in page_names if not _check_valid_page_name(pagename=p)]
+    if len(invalid_pages) > 0:
```

Comment:
Empty lists are identical to `None`, which evaluates in lists as `False`.

As a result, you can instead just say `if invalid_pages:`

Reply:
Thanks for the spot, change all occurrences to better syntax